core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMap.java

```diff
@@ -26,7 +26,7 @@
 /**
  * Datamap is an entity which can store and retrieve index data.
  */
-public interface DataMap {
+public interface DataMap<T extends Blocklet> {
```

Comment:
If there are only FG and CG datamap, I prefer to make another interface for FG datamap instead of adding generic. So there will be CoarseGrainDataMap and FineGrainDataMap interface

Reply:
Yes, but methods are same for both, so I have added Abstract classes for CG and FG to distinguish between them.